### DISCUSSION TITLE
Persist localStorage wihout --location when upgrading installed scripts?

### DISCUSSION THREAD

**Discussion by ekmwest** [denoland/deno#14616] [Q&A] [answered]
The `--location` flag will persist localStorage over script upgrades:

```bash
# INSTALL v0.0.1
$ deno install --location=<URL> <SERVER>@0.0.1/test.js

$ test
localStorage is empty

$ test
localStorage has data


# UPGRADE to v0.0.2
$ deno install -f --location=<URL> <SERVER>@0.0.2/test.js

$ test
localStorage has data
```

Example script:
```js
// test.js
#!/usr/bin/env -S deno run
console.log(localStorage.getItem('test') || 'localStorge is empty');
localStorage.setItem('test', 'localStorage has data');
```

But is it possible to persist localStorage ***without*** using the `--location` flag?

Thank you.

**Comment by nayeemrmn** [comment, marked as answer]
It's not possible, this is an intended use of `--location`. It's not considered too much of a burden to specify it because `deno install` commands are typically copy-pasted from your project README once* and it won't have to be specified by your users beyond that.

*Well, more than once if we consider upgrades. Some installed programs provide an `upgrade` subcommand which does this for you, e.g. https://github.com/denoland/deployctl/blob/eb9f6f71747443b9dbd074d2c417e7e94df576be/src/subcommands/upgrade.ts.

**Reply by ekmwest** [reply]
Thank you! 🙏
